suricata/update/main.py

```diff
@@ -352,6 +352,7 @@ def get_tmp_filename(self, url):
 
     def fetch(self, url):
         net_arg = url
+        no_checksum = url[2] if isinstance(url, tuple) else False
```

Comment:
Its usually nice to avoid these kinds of checks i.e. maintain the type of data all across the code so there is consistency.

Reply:
@shivan1b I'm not sure if I understand this. Could you please elaborate more?

Reply:
@vagishagupta23 checking multiple times in a code the "type" of a variable is usually not a good practice. You declare something as string then it should be string wherever its referenced, it should not be a tuple at some places and a string at others. Brings in data inconsistency and chances of code crashing increase.

Reply:
@shivan1b Does that mean I should probably use some other check in place of this? But that seems difficult as url can possibly be tuple sometimes and string when added though the command line.
Or maybe instead of checking the type twice I could check it once, something like:
```
if isinstance(url, tuple):
    no_checksum = url[2]
    url = url[0]
else:
    no_checksum = False
    url = url
```
What are your thoughts on it?

Reply:
@vagishagupta23 I meant that the internal functions should receive a consistent data type. As in if url is supposed to be a tuple, it should be passed as a tuple from every place that calls the inner function. Then we know that `fetch` always expects urls to be tuples and put checks accordingly inside the function.
Does this make things clear?

Reply:
@shivan1b Ok, understood.